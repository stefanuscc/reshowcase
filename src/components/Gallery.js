import GalleryItem from './micro/GalleryItem'
import GallerySkel from './micro/GallerySkel'

export const Gallery = (props) => {
    const { images, isLoading, ...restProps } = props

    return (
        <div {...restProps} className="grid grid-cols-1 gap-y-5 sm:grid-cols-2 gap-x-3 lg:grid-cols-2 xl:grid-cols-5 xl:gap-x-3">
            { isLoading || images == null ? '' : images.map((image, idx) => (
                <GalleryItem image={image} idx={idx} key={image.id}></GalleryItem>
            ))}
            { isLoading || images == null ? [...Array(5)].map((e, i) => (
                <GallerySkel key={i} idx={i}></GallerySkel>
            )) : ''}
        </div>
    )
}

export default Gallery