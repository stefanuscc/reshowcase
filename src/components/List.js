import ListSkel from './micro/ListSkel'
import ListItem from './micro/ListItem'

export const List = (props) => {
    const { images, isLoading } = props

    return (
        <div className="bg-white shadow overflow-hidden sm:rounded-md">
            <ul className="divide-y divide-gray-200">   
                { isLoading || images == null ? [...Array(5)].map((e, i) => (
                    <ListSkel key={i} idx={i}/>
                )) : ''}
                { isLoading || images == null ? '' : images.map((image, idx) => (
                    <ListItem key={image.id} image={image} idx={idx}/>
                ))}
            </ul>
        </div>
    )
}

export default List