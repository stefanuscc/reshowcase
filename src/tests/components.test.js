import { cleanup } from "@testing-library/react";
import { Gallery } from "../components/Gallery";
import { List } from "../components/List";

import renderer from "react-test-renderer"

afterEach(cleanup);

it("Renders proper loading gallery mode", () => {
    const view = renderer.create(<Gallery isLoading={true} images={[]}/>);

    expect(view).toMatchSnapshot();
});

it("Renders proper loading list mode", () => {
    const view = renderer.create(<List isLoading={true} images={[]}/>);

    expect(view).toMatchSnapshot();
});