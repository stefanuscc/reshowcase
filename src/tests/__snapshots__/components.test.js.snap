// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders proper loading gallery mode 1`] = `
<div
  className="grid grid-cols-1 gap-y-5 sm:grid-cols-2 gap-x-3 lg:grid-cols-2 xl:grid-cols-5 xl:gap-x-3"
>
  
  <div
    className="animate-pulse flex-row"
  >
    <div
      className="w-full aspect-w-5 aspect-h-3 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-4 bg-slate-200"
    />
    <div
      className="mt-4 h-5 w-full bg-slate-200 rounded"
    />
    <div
      className="mt-1 h-6 w-full bg-slate-200 rounded"
    />
  </div>
  <div
    className="animate-pulse flex-row"
  >
    <div
      className="w-full aspect-w-5 aspect-h-3 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-4 bg-slate-200"
    />
    <div
      className="mt-4 h-5 w-full bg-slate-200 rounded"
    />
    <div
      className="mt-1 h-6 w-full bg-slate-200 rounded"
    />
  </div>
  <div
    className="animate-pulse flex-row"
  >
    <div
      className="w-full aspect-w-5 aspect-h-3 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-4 bg-slate-200"
    />
    <div
      className="mt-4 h-5 w-full bg-slate-200 rounded"
    />
    <div
      className="mt-1 h-6 w-full bg-slate-200 rounded"
    />
  </div>
  <div
    className="animate-pulse flex-row"
  >
    <div
      className="w-full aspect-w-5 aspect-h-3 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-4 bg-slate-200"
    />
    <div
      className="mt-4 h-5 w-full bg-slate-200 rounded"
    />
    <div
      className="mt-1 h-6 w-full bg-slate-200 rounded"
    />
  </div>
  <div
    className="animate-pulse flex-row"
  >
    <div
      className="w-full aspect-w-5 aspect-h-3 rounded-lg overflow-hidden xl:aspect-w-7 xl:aspect-h-4 bg-slate-200"
    />
    <div
      className="mt-4 h-5 w-full bg-slate-200 rounded"
    />
    <div
      className="mt-1 h-6 w-full bg-slate-200 rounded"
    />
  </div>
</div>
`;

exports[`Renders proper loading list mode 1`] = `
<div
  className="bg-white shadow overflow-hidden sm:rounded-md"
>
  <ul
    className="divide-y divide-gray-200"
  >
    <li
      className="animate-pulse"
    >
      <div
        className="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          className="min-w-0 flex-1 flex items-center"
        >
          <div
            className="flex w-1/5"
          >
            <div
              className="w-full aspect-w-2 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-2 xl:aspect-h-1 bg-slate-200"
            />
          </div>
          <div
            className="min-w-0 flex-1 px-4 md:grid md:grid-cols-3 md:gap-4"
          >
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            className="h-5 w-5 bg-slate-200 rounded"
          />
        </div>
      </div>
    </li>
    <li
      className="animate-pulse"
    >
      <div
        className="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          className="min-w-0 flex-1 flex items-center"
        >
          <div
            className="flex w-1/5"
          >
            <div
              className="w-full aspect-w-2 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-2 xl:aspect-h-1 bg-slate-200"
            />
          </div>
          <div
            className="min-w-0 flex-1 px-4 md:grid md:grid-cols-3 md:gap-4"
          >
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            className="h-5 w-5 bg-slate-200 rounded"
          />
        </div>
      </div>
    </li>
    <li
      className="animate-pulse"
    >
      <div
        className="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          className="min-w-0 flex-1 flex items-center"
        >
          <div
            className="flex w-1/5"
          >
            <div
              className="w-full aspect-w-2 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-2 xl:aspect-h-1 bg-slate-200"
            />
          </div>
          <div
            className="min-w-0 flex-1 px-4 md:grid md:grid-cols-3 md:gap-4"
          >
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            className="h-5 w-5 bg-slate-200 rounded"
          />
        </div>
      </div>
    </li>
    <li
      className="animate-pulse"
    >
      <div
        className="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          className="min-w-0 flex-1 flex items-center"
        >
          <div
            className="flex w-1/5"
          >
            <div
              className="w-full aspect-w-2 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-2 xl:aspect-h-1 bg-slate-200"
            />
          </div>
          <div
            className="min-w-0 flex-1 px-4 md:grid md:grid-cols-3 md:gap-4"
          >
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            className="h-5 w-5 bg-slate-200 rounded"
          />
        </div>
      </div>
    </li>
    <li
      className="animate-pulse"
    >
      <div
        className="flex items-center px-4 py-4 sm:px-6"
      >
        <div
          className="min-w-0 flex-1 flex items-center"
        >
          <div
            className="flex w-1/5"
          >
            <div
              className="w-full aspect-w-2 aspect-h-1 bg-gray-200 rounded-lg overflow-hidden xl:aspect-w-2 xl:aspect-h-1 bg-slate-200"
            />
          </div>
          <div
            className="min-w-0 flex-1 px-4 md:grid md:grid-cols-3 md:gap-4"
          >
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
            <div>
              <p
                className="h-4 w-full bg-slate-200 rounded"
              />
              <p
                className="mt-2 h-5 w-full bg-slate-200 rounded"
              />
            </div>
          </div>
        </div>
        <div>
          <p
            className="h-5 w-5 bg-slate-200 rounded"
          />
        </div>
      </div>
    </li>
    
  </ul>
</div>
`;
